chain/chainhandle.go

```diff
@@ -1035,7 +1032,8 @@ func executeTx(execCtx context.Context, ccc consensus.ChainConsensusCluster, cdb
 	receipt.TxHash = tx.GetHash()
 	receipt.Events = events
 	receipt.FeeDelegation = txBody.Type == types.TxType_FEEDELEGATION
-	receipt.GasUsed = contract.GasUsed(txFee, bs.GasPrice, txBody.Type, bi.ForkVersion)
+	isGovernance := txBody.Type == types.TxType_GOVERNANCE
+	receipt.GasUsed = fee.ReceiptGasUsed(bi.ForkVersion, isGovernance, txFee, bs.GasPrice)
```

Comment:
I prefer passing tx type here

Reply:
fee package can`t import tx type because of cycling import error. ( fee <-> types )
( in my opinion, it is weird that types package import fee packages. )

Reply:
OK, no problem

Reply:
it seems validate tx logic ( in types package ) is not clean. it can be resolved after refactoring validate tx logic.

Reply:
please let this for the next year. V4 is more important!

Reply:
ok. i`ll do it later.